src/nvim/api/buffer.c

```diff
@@ -1571,45 +1573,42 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id, Integer line, Integer
     return 0;
   }
 
-
   uint64_t id = 0;
   if (opts->id.type == kObjectTypeInteger && opts->id.data.integer > 0) {
     id = (uint64_t)opts->id.data.integer;
   } else if (HAS_KEY(opts->id)) {
       api_set_error(err, kErrorTypeValidation, "id is not a positive integer");
-      goto error;
+      return 0;
```

Comment:
Yes, but before line 1620, the call to `clear_virttext` should not be necessary. If we value clarity, it would be better to change `return 0;` to `goto error:`. If the cost of the function call is an issue, I think the `goto error;` in line 1615 should also be `return 0;`. I don't really have an opinion on this, so I'd like to hear @bfredl opinion.

https://github.com/neovim/neovim/blob/74c73bf20502057f17c68375af569d5ae4e1a11f/src/nvim/api/buffer.c#L1612-L1620

Reply:
Besides clarity, it helps avoid future errors, where cleanup step is added to `error`, but these `return` areas need to be re-evaluated to see if they need to be updated. 

> I'd like to hear @bfredl opinion.

sure.

Reply:
I'd say use `goto error` everywhere for consistency. an extra function call in the _error_ path is not a hot-path cost.